pkg/kubectl/cmd/apply.go

```diff
@@ -327,13 +327,14 @@ func RunApply(f cmdutil.Factory, cmd *cobra.Command, out, errOut io.Writer, opti
 			if err != nil {
 				return cmdutil.AddSourceToErr(fmt.Sprintf("applying patch:\n%s\nto:\n%v\nfor:", patchBytes, info), info.Source, err)
 			}
+			if string(patchBytes) != "{}" {
```

Comment:
Do you need to check for the empty string here too?

Reply:
I think no empty string will occur, beacuse patchBytes are generated by `json.Marshal`